src/utils/warn.spec.js

```diff
@@ -0,0 +1,33 @@
+import warn from './warn'
+
+describe('utils/warn', () => {
+  const dummyWarning = 'A Rush Of Blood To The Head'
+
+  beforeAll(() => {
+    jest.spyOn(console, 'warn').mockImplementation(() => {})
+  })
+
+  describe('with BOOTSTRAP_VUE_NO_WARN environment variable set', () => {
+    beforeAll(() => {
+      process.env.BOOTSTRAP_VUE_NO_WARN = true
+    })
+
+    it('does not call console.warn()', () => {
+      warn(dummyWarning)
+
+      expect(console.warn).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('without BOOTSTRAP_VUE_NO_WARN environment variable set', () => {
+    beforeAll(() => {
+      global.process = null
```

Comment:
This seems quite unfriendly. Do we want

```suggestion
      process.env.BOOTSTRAP_VUE_NO_WARN = undefined
```

instead?

Reply:
This would assume `process.env` exists (assigning to a nested property on an object that might not exist).

I would also switch to `beforeEach`, and create an `afterEach` to restore the value to what ever it was originally.

```js
describe('...', () => {
  process.env = process.env || {}
  oldEnv = { ...process.env }
  beforeEach(() => {
    // Alter the environment
    process.env.BOOTSTRAP_VUE_NO_WARN = undefiend
  })
  afterEach(() =>
    // Restore the environment
    process.env = oldEnv
  })
  it('...', async () => {
    // test here
  })
})
```

And if you need to test with both set and unset environment, create two describe sections, each with their own beforeEach and afterEach sections

Reply:
> and create an afterEach to restore the value to what ever it was originally.

Makes sense! 👍 I'll restore the complete `process` object though (because restoring `process.env` would fail after `process = null`).

Reply:
Note, you will need to do a deep clone, as nested objects in process would be affected.

Reply:
I'm mocking the process object completely, so I don't think we need to clone the original one. I have just pushed my changes, so please let me know if you disagree. 😃

Reply:
> I don't think we need to clone the original one

...in addition as far as I know, the `process` object is not JSON serializable, so deep cloning would be difficult.